compiler/rustc_builtin_macros/src/errors.rs

```diff
@@ -369,21 +369,21 @@ pub(crate) struct DerivePathArgsValue {
 }
 
 #[derive(Diagnostic)]
-#[diag(builtin_macros_no_default_variant)]
-#[help]
+#[diag(builtin_macros_no_default_variant, code = E0665)]
 pub(crate) struct NoDefaultVariant {
     #[primary_span]
     pub(crate) span: Span,
+    #[label]
+    pub(crate) item_span: Span,
     #[subdiagnostic]
     pub(crate) suggs: Vec<NoDefaultVariantSugg>,
 }
 
-#[derive(Subdiagnostic)]
+#[derive(Subdiagnostic, Debug)]
 #[suggestion(
     builtin_macros_suggestion,
     code = "#[default] {ident}",
```

Comment:
Preexisting: Ideally, this suggestion would be `(v.span.shrink_to_hi(), "#[default] ")` instead of `(v.span, format!("#[defaukt] {ident}"))` like it is right now. This would also allow us to get rid of the `pub(crate) ident: Ident,`.

If you want to make the change feel free to do so in this PR :)